docs/examples/ex_extract.rst

```diff
@@ -0,0 +1,515 @@
+                sel_name=["dummy_drifts", "altitude"])
+
+    # CPU times: user 419 ms, sys: 13 ms, total: 432 ms
+    # Wall time: 431 ms
+
+    # Print results from interpolation
+    inst['model_dummy_drifts']
+
+    Epoch
+    2009-01-01 00:00:00    22.393249
+    2009-01-01 00:00:01    22.405926
+    2009-01-01 00:00:02    22.418600
+    2009-01-01 00:00:03    22.431272
+    2009-01-01 00:00:04    22.443941
+                             ...
+    2009-01-01 00:01:35    23.592833
+    2009-01-01 00:01:36    23.605252
+    2009-01-01 00:01:37    23.617668
+    2009-01-01 00:01:38    23.630081
+    2009-01-01 00:01:39    23.642492
+    Name: model_dummy_drifts, Length: 100, dtype: float64
+
+In the interpolation function, ``inst`` and :py:attr:`model.data` provide
+the required data through the :py:class:`pysat.Instrument` and
+:py:class:`xarray.Dataset` objects. The ::
+
+   ["altitude", "latitude", "longitude"]
+
+term provides the content and ordering of the spatial locations for
+``inst``. The subsequent ::
+
+   ["ilev", "latitude", "longitude"]
+
+term provides the equivalent regular dimension labels from
+:py:attr:`model.data`, in the same order as the underlying model dimensions.
+While this function does operate on irregular data it also needs information on
+the underlying regular memory structure of the variables. The ::
+
+   "time"
+
+terms cover the model label used for the datetime coordinate. The ::
+
+   ["cm", "deg", "deg"]
+
+term covers the units for the model information (altitude/latitude/longitude)
+that maps to the ``inst`` information in the coordinate list
+``["altitude", "latitude", "longitude"]``. Note that the ``"cm"``
+covers units for ``'altitude'`` in :py:attr:`model.data`, the variable
+that will replace ``'ilev'``, while the second two list elements (both
+``"deg"``) covers the units for the latitude and longitude dimensions.
+Units for the corresponding information from ``inst`` are taken directly
+from the :py:class:`pysat.Instrument` object. The ::
+
+    "ilev"
+
+identifies the regular model dimension that will be replaced with irregular
+data for interpolation. The ::
+
+    "altitude"
+
+identifies the irregular model variable that will replace the regular
+coordinate. The ::
+
+    [50., 10., 10.]
+
+term is used to define a half-window for each of the ``inst`` locations,
+in units from ``inst``, used to downselect data from :py:attr:`model.data`
+to reduce computational requirements. In this case a window of +/-50 km in
+altitude, +/-10 degrees in latitude, and +/-10 degrees in longitude is used.
+The keyword argument ::
+
+    sel_name = ["dummy_drifts", "altitude"]
+
+identifies the :py:attr:`model.data` variables that will be interpolated onto
+``inst``.  If you don't account for the irregularity in the desired
+model coordinates, the interpolation results are affected.
+
+.. code:: python
+
+    new_data_keys = pysatModels.utils.extract.instrument_view_through_model(
```

Comment:
This fails for me with the error: "ValueError: Unknown instrument location index model_pressure"

I think this is because on lines 401 - 408 a new instrument is instantiated but the "instrument_altitude_to_model_pressure" is not used so the model_pressure index isn't available at this point. Perhaps I misreaad the instructions and I wasn't meant to move linearly through these instructions. Or maybe I missed a chunk of the example.

Reply:
Assigning @rstoneback to make the change.

Reply:
I  added line 641 which accounts for the missing variable via instrument_altitude_to_model_pressure some time back but forgot to hit resolve. I also had to reload multiple times today to get github to show it.

Reply:
To help clear up potential confusion.

Github still shows all these comments mapping to line 492 via the main pull page, but with persistence you can get the github file browser to show up these comment boxes are now really down in the 600s. Not sure what the trick is. Mostly when I click through starting from here GitHub shows the same specific and old commit.

For confirmation of the current state of the file, click and scroll: https://github.com/pysat/pysatModels/blob/rc_0.1.0/docs/examples/ex_extract.rst

🐬